mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -1037,7 +1037,7 @@ def get_latest_metrics_records(
                 filter_query=filter_query,
                 timestamp_column=mm_schemas.WriterEvent.END_INFER_TIME,
                 # Aggregate per application/metric pair regardless of timestamp
-                group_by=columns[1:],
+                group_by=columns[1:-1],
```

Comment:
this `[1:-1]` is kind of sneaky/fragile.
better do in the if/else clause: 
```
group_by = [mm_schemas.ResultData.RESULT_STATUS,
                    mm_schemas.ResultData.RESULT_KIND,]
columns = group_by + [mm_schemas.ResultData.RESULT_NAME, mm_schemas.ResultData.RESULT_VALUE]
```
or just build group_by columns separately